src/lib/components/ScanAnNft.svelte

```diff
@@ -0,0 +1,167 @@
+<script lang="ts">
+  import { onMount } from 'svelte';
+  import env from '$lib/constants/env';
+
+  let hasEvaluatedNft: boolean = false;
+  let openseaUrl: string = '';
+  let imageUrl: string = '';
+  let contractAddress: string = '';
+  let tokenId: string = '';
+
+  function parseAndSetFromOpenseaUrl(url) {
+    parseValuesFromOpenseaUrl(url);
+    fetchImage(contractAddress, tokenId);
+  }
+
+  function parseValuesFromOpenseaUrl(url: string) {
+		// https://opensea.io/assets/:nftContractAddress/:tokenId
+		let openseaAddressAndIdRe = /https:\/\/opensea.io\/assets\/(\w*)\/(\w*)/g;
+		let [_, address, id] = openseaAddressAndIdRe.exec(url);
+
+		if (address && id) {
+			openseaUrl = url;
+			contractAddress = address;
+			tokenId = id;
+		}
+  }
+
+  async function fetchImage(_contractAddress: string, _tokenId: string): Promise<void> {
+    console.log("fetchImage()");
```

Comment:
You may want to reset the variables you mutate in the function upon invocation. Could optionally put it in a loading state.